test/relations.test.js

```diff
@@ -819,6 +819,18 @@ describe('relations', function() {
             done();
           });
         });
+        it('error case: filtering not done with wrong id key', function(done) {
+          var wrongWhereFilter = {where: {wrongIdKey: samplePatientId}};
+          physician.patients(wrongWhereFilter, function(err, ch) {
+            if (err) return done(err);
+            should.exist(ch);
+            ch.should.have.lengthOf(3);
```

Comment:
>  According to @jannyHou, juggler or some connectors act differently.
 
Sorry ignore what I said, it seems at least memory and cloudant have same behaviour.

@dhmlau I give a quick try with both memory and cloudant connector:
- {where: {wrongKey: "wrong"}} - returns [ ]
- {wrongKey: "wrong"}  - returns all records (DANGEROUS IMO)(not related to this PR though)

So I am afraid the test case here is not correct. Even it's a relation api, juggler generates the filter to targetModel as 
`where: {and: [{id: {inq: [fkValue_array]}}, {wrongKey: 'somevalue'}]}`, which should not return any result.